Sources/SwiftParser/Declarations.swift

```diff
@@ -700,7 +704,7 @@ extension Parser {
             arena: self.arena
           )
         )
-      } while keepGoing != nil && self.hasProgressed(&loopProgress)
+      } while keepGoing != nil && self.hasProgressed(&loopProgress) && !self.atWhereClauseListTerminator()
```

Comment:
Generic where clauses don’t necessarily need to be followed by a brace. For example you can have the following, which is valid.

```swift
protocol Foo {
  func test<T>(abc: T) where T: Comparable
}
```

Because of the lack of a proper terminated for the where classes, I’m actually not sure if we should allow trailing commas here. Checking if the next token is a keyword also doesn’t work because it could be a contextual keyword like `nonisolated` on the following function declaration.